docs/naming_conventions.md

```diff
@@ -0,0 +1,71 @@
+
+# Naming Conventions in Kids First FHIR Data Model
+
+This document describes the naming conventions for ids and
+the file names for resources in the Kids First FHIR data model.
+
+## Resource Id
+
+1. A resource in the FHIR model should always have the `id` attribute
+populated by model developers. We should never let the server auto populate
+`id` since it is the primary way that machines and humans will remember,
+reference, and lookup a resource.
+
+2. The `id` value should conform to the the regex pattern in the
+[FHIR spec](https://www.hl7.org/fhir/datatypes.html#id).
+
+3. All resource `id` values should have the same casing. There does not seem
+   to be a consistent standard or recommendation for this but we should pick
+   one and stick to it. Based on observations, the options are:
+   - Camel case
+   - Snake case with hyphens instead of underscores
+
+TODO - Choose casing
+
+## Resource File Name
+
+1. All resource types EXCEPT `StructureDefinition` should follow this
```

Comment:
I don't understand the benefit of having two different formats. 
`<thing type><delim><base type><delim><resource id>.json` works equally well for both and preserves the bullet points.